jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -1284,7 +1284,7 @@ public void setTrailerFields(Supplier<Map<String, String>> trailers)
 
     protected MetaData.Response newResponseMetaData()
     {
-        return new MetaData.Response(_channel.getRequest().getHttpVersion(), getStatus(), getReason(), _fields, getLongContentLength(), getTrailers());
+        return new MetaData.Response(_channel.getRequest().getHttpVersion(), getStatus(), getReason(), _fields, getLongContentLength(), getTrailers(), getHttpChannel().isServletUpgrade());
```

Comment:
Again, can you explain why the generator needs to know that it is a servlet upgrade?   It should always be able to send 1xx responses and continue.      It is probably OK to assume that a 101 will always go to infinite content mode, as mostly the HttpGenerator will be replaced and it is only the servlet upgarde model that it is not.    So it can probably just assume servlet upgrade and after the 101 be happy to send any content it is given as pure bytes without any framing.

Reply:
I modified the impl according to our discussion on the matter.